components/definition/cellRules/sections/NetPyNESectionThumbnail.js

```diff
@@ -11,59 +16,50 @@ export default class NetPyNESectionThumbnail extends React.Component {
     this.state = {
       isHovered: false,
       dialogOpen: false
-    }
-
-    this.handleClick = this.handleClick.bind(this);
-    this.handleHoverIn = this.handleHoverIn.bind(this);
-    this.handleHoverOut = this.handleHoverOut.bind(this);
-    this.handleDialogBox = this.handleDialogBox.bind(this);
-  }
+    };
+  };
 
   handleClick() {
-    if (this.props.handleClick) {
-      if(this.props.selected && this.state.isHovered) {
-        this.setState({dialogOpen: true});
-      } else {
-        this.props.handleClick(this.props.name, true);
+    const { isHovered } = this.state;
+    const { name, handleClick, selected } = this.props;
+    if (handleClick) {
+      selected && isHovered ? this.setState({dialogOpen: true}) : handleClick(name, true);
       }
-    }
-  };
-
-  handleHoverIn() {
-    this.setState({
-      isHovered: true
-    });
-  };
-
-  handleHoverOut() {
-    this.setState({
-      isHovered: false
-    });
   };
 
   handleDialogBox(response) {
-    if(this.props.handleClick && response) {
-      this.props.deleteMethod(this.props.name);
+    const { name, handleClick, deleteMethod } = this.props;
+    if (handleClick && response) {
+      deleteMethod(name);
     }
     this.setState({dialogOpen: false});
   };
 
   render() {
+    const { name, selected } = this.props;
+    const { isHovered, dialogOpen } = this.state;
     return (
       <div>
       <RaisedButton
-        id={this.props.name}
-        onMouseEnter={this.handleHoverIn}
-        onMouseLeave={this.handleHoverOut}
+        id={name}
         primary={true} 
-        className={"rectangularActionButton " + (this.props.selected ? "selectedRectangularActionButton " : "")} 
-        onClick={this.handleClick}>
-          {(this.state.isHovered && this.props.selected) ? <FontIcon className="fa fa-trash-o" color="white" hoverColor="white"/> : this.props.name}
+        style={ styles.btn }
+        buttonStyle={ styles.btn }
+        onMouseEnter={ () => this.setState({isHovered: true}) }
+        onMouseLeave={ () => this.setState({isHovered: false}) }
+        data-tooltip={isHovered && name.length > 14 ? name : undefined}
+        className={"rectangularActionButton " + (selected ? "selectedRectangularActionButton " : "")} 
+        onClick={() => this.handleClick()}
+      >
+        {(isHovered && selected)
```

Comment:
Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?